zerver/lib/actions.py

```diff
@@ -802,6 +805,14 @@ def do_send_messages(messages):
         if message['sender_queue_id'] is not None:
             event['sender_queue_id'] = message['sender_queue_id']
         send_event(event, users)
+
+        if links_for_embed:
+            event_data = {
+                'message_id': message['message'].id,
+                'message_content': message['message'].content,
+                'urls': links_for_embed}
+            queue_json_publish('embed_links', event_data, lambda x: None)
```

Comment:
I would probably check the setting here, and not run this code path if it isn't enabled, for clarity if nothing else.